### PR TITLE
feat: Add support for `spot-instance-requests` tagging for launched instances

### LINKED ISSUE #4550: Add tags to spot requests created by Karpenter
### Description

**What problem are you trying to solve?**

Currently, when customers create spot requests using Karpenter, the spot requests do not have any tags specified, while other resources like instances have the correct tags applied. 

This issue was discussed offline with @jonathan-innis, and it appears that adding tags to spot requests created by Karpenter is not supported due to limitations with EC2 APIs. These limitations prevent the tag specifications from being passed down to the RunInstances API.

Documentation Reference:

https://karpenter.sh/preview/concepts/settings/#awstags
https://karpenter.sh/docs/concepts/node-templates/#spectags

**How important is this feature to you?**

This issue has been raised on behalf of a customer, highlighting the importance of this feature. Tags are essential for resource management and tracking within AWS, and having this functionality in Karpenter would greatly enhance the user experience.

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by jonathan-innis**
Fixes #4550

**Description**

This PR adds tags to the `spot-instance-requests` resource type when launching instances so that these resources are scoped to Karpenter and are easily tracked by users who are tracking resources through tags.

**How was this change tested?**

`make presubmit`

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

File: website/content/en/preview/upgrading/upgrade-guide.md [hunk 1]
```diff
@@ -32,6 +32,10 @@ kubectl apply -f https://raw.githubusercontent.com/aws/karpenter{{< githubRelRef
 kubectl apply -f https://raw.githubusercontent.com/aws/karpenter{{< githubRelRef >}}pkg/apis/crds/karpenter.k8s.aws_ec2nodeclasses.yaml
 ```
 
+### Upgrading to v0.33.0+
+
+* Karpenter now tags `spot-instances-request` with the same tags that it tags instances, volumes, and primary ENIs. Practically, this means that you can further scope your controller policy for the `ec2:RunInstances` and `ec2:CreateTags` actions to require that it launches the `spot-instance-request` with specific tags. You can view an example of scoping these actions in the [Getting Started Guide's default CloudFormation controller policy](https://github.com/aws/karpenter/blob/v0.33.0/website/content/en/preview/getting-started/getting-started-with-karpenter/cloudformation.yaml#L61).
```
Comment (njtran) [inline website/content/en/preview/upgrading/upgrade-guide.md:37]: I wouldn't really call this out as an breaking change, as I think this section is usually reserved for. That being said, I don't mind it being here.

**Review by njtran — APPROVED** [review]

File: website/content/en/preview/upgrading/upgrade-guide.md (hunk 1 above)
Comment (jonathan-innis) [reply website/content/en/preview/upgrading/upgrade-guide.md:37]: Yeah, I don't think that we've necessarily made this just "breaking changes". This functions as release notes IMO so info like this makes sense here.

**Pull request merged**